internal/commands/list.go

```diff
@@ -0,0 +1,96 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/deislabs/duffle/pkg/claim"
+
+	"github.com/docker/app/internal/store"
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	units "github.com/docker/go-units"
+	"github.com/spf13/cobra"
+)
+
+type listOptions struct {
+	targetContext string
+}
+
+var (
+	listColumns = []struct {
+		header string
+		value  func(c *claim.Claim) string
+	}{
+		{"INSTALLATION", func(c *claim.Claim) string { return c.Name }},
+		{"ACTION", func(c *claim.Claim) string { return c.Result.Action }},
+		{"RESULT", func(c *claim.Claim) string { return c.Result.Status }},
+		{"CREATED", func(c *claim.Claim) string { return units.HumanDuration(time.Since(c.Created)) }},
+		{"MODIFIED", func(c *claim.Claim) string { return units.HumanDuration(time.Since(c.Modified)) }}}
+)
+
+func listCmd(dockerCli command.Cli) *cobra.Command {
+	var opts listOptions
+
+	cmd := &cobra.Command{
+		Use:     "list [OPTIONS]",
+		Short:   "List the installations and their last known installation result",
+		Aliases: []string{"ls"},
+		Args:    cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runList(dockerCli, opts)
+		},
+	}
+	cmd.Flags().StringVar(&opts.targetContext, "target-context", "", "List installations on this context")
```

Comment:
Should we provide a `--format` like list actions in the docker CLI ?

Reply:
Maybe in a follow-up